azkaban-web-server/src/main/java/azkaban/webapp/servlet/LoginAbstractAzkabanServlet.java

```diff
@@ -121,7 +121,7 @@ protected void doGet(final HttpServletRequest req, final HttpServletResponse res
 
       handleGet(req, resp, session);
     } else {
-      if (hasParam(req, "ajax")) {
```

Comment:
I agree that it's better this way, but this is a breaking change on our api: anyone using our apis outside the website without specifying the header 'X-Requested-With' (not enforced) in POST requests will start getting json instead of html as a response when session expires which can happen often. Api versioning is the way to conduct these type of changes but unfortunately we haven't planned for that yet. I don't know what the impact of this change is but I prefer not to risk it.